tests/Unit/Evolution/Systems/Cce/Actions/Test_CalculateScriInputs.cpp

```diff
@@ -0,0 +1,287 @@
+#include "NumericalAlgorithms/Spectral/SwshFiltering.hpp"
+#include "NumericalAlgorithms/Spectral/SwshTransform.hpp"
+#include "ParallelAlgorithms/Actions/MutateApply.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/GeneralRelativity/KerrSchild.hpp"
+#include "Time/Tags.hpp"
+#include "Time/TimeSteppers/RungeKutta3.hpp"
+#include "Time/TimeSteppers/TimeStepper.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/VectorAlgebra.hpp"
+#include "tests/Unit/ActionTesting.hpp"
+#include "tests/Unit/Evolution/Systems/Cce/BoundaryTestHelpers.hpp"
+#include "tests/Unit/NumericalAlgorithms/Spectral/SwshTestHelpers.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+#include "tests/Utilities/MakeWithRandomValues.hpp"
+
+namespace Cce {
+
+namespace {
+// these will be generated in the test
+using all_pre_swsh_derivative_scri_input_tags = tmpl::list<
+    Tags::BondiH, Tags::DuRDividedByR, Tags::EthRDividedByR,
+    Tags::Dy<Tags::BondiJ>, Tags::Dy<Tags::BondiW>, Tags::Dy<Tags::BondiQ>,
+    Tags::Dy<Tags::BondiU>, Tags::Dy<Tags::Dy<Tags::BondiBeta>>,
+    Spectral::Swsh::Tags::Derivative<Tags::BondiBeta,
+                                     Spectral::Swsh::Tags::EthEthbar>>;
+
+// these need to be computed via `PreSwshDerivatives` and
+// `PrecomputeCceDependencies`, but are not calculated in the action being
+// tested
+using extra_pre_swsh_derivative_scri_tags =
+    tmpl::list<Tags::Dy<Tags::Dy<Tags::BondiJ>>>;
+
+using extra_precomputation_scri_tags = tmpl::list<Tags::OneMinusY>;
+
+using all_real_boundary_scri_input_tags =
+    tmpl::list<Tags::InertialRetardedTime>;
+
+using transform_buffers_for_scri =
+    tmpl::remove_duplicates<tmpl::flatten<tmpl::transform<
+        all_swsh_derivative_tags_for_scri,
+        tmpl::bind<Spectral::Swsh::coefficient_buffer_tags_for_derivative_tag,
+                   tmpl::_1>>>>;
+
+template <typename Metavariables>
+struct mock_characteristic_evolution {
+  using component_being_mocked = CharacteristicEvolution<Metavariables>;
+  using replace_these_simple_actions = tmpl::list<>;
+  using with_these_simple_actions = tmpl::list<>;
+
+  using simple_tags = db::AddSimpleTags<
+      ::Tags::Variables<tmpl::append<all_swsh_derivative_tags_for_scri,
+                                     all_pre_swsh_derivative_tags_for_scri,
+                                     all_pre_swsh_derivative_scri_input_tags,
+                                     extra_pre_swsh_derivative_scri_tags,
+                                     extra_precomputation_scri_tags>>,
+      ::Tags::Variables<transform_buffers_for_scri>,
+      ::Tags::Variables<all_real_boundary_scri_input_tags>,
+      ::Tags::Variables<
+          tmpl::append<all_boundary_pre_swsh_derivative_tags_for_scri,
+                       all_boundary_swsh_derivative_tags_for_scri>>>;
+  using compute_tags = db::AddComputeTags<>;
+
+  using initialize_action_list =
+      tmpl::list<ActionTesting::InitializeDataBox<simple_tags, compute_tags>>;
```

Comment:
I'll inline the `compute_tags`, but I think it's more readable to split the `simple_tags` into a separate action list to avoid too many nested angle brackets.

Reply:
since `InitializeDataBox` defaults to an empty `ComputeTagsList` you don't even need that :)